### ISSUE TITLE
Ensure the contrast between foreground and background colors meets WCAG 2 AA minimum contrast ratio thresholds (#basic\.label): A11y_Microsoft MakeCode_New projet_Editor_AI4W

### ISSUE THREAD

**Issue by kupatkar99** [microsoft/pxt-microbit#6832]
"Leveraging SWE Agent can help accelerate this accessibility bug fix; review and validate all AI generated changes before merge."

"Try [ES Chat](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2feng.ms%2fdocs%2fcoreai%2fdevdiv%2fone-engineering-system-1es%2f1es-docs%2fes-chat%2findex&data=05%7c02%7cv-mchougule%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838215421556%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=%2B1yeHqbNhHcMuQaZNXDo%2B%2fiWxH0mZFSmWpamQk2ftws%3D&reserved=0) to learn more about the MAS rule and how to fix the issue. If you need more help, use our [Teams channel](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fteams.microsoft.com%2fl%2fchannel%2f19%253ADeCtYeREyQHw5jYsz9PPc3hpVw48m-tOlTkQ1jEn0eM1%2540thread.tacv2%2fCAI%2520Accessibility%2520Support%2520Channel?groupId%3Df5eb428c-9e6e-4c16-8143-33fe3961a9ae%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=05%7c02%7cv-mchougule%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838215450064%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=yncZXSKXaJlQ3HpXeR3TK03MNIhUe8tV2yZl49clFZY%3D&reserved=0) or [office hours](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2foutlook.office.com%2fbook%2fmicrosoft2779007%2540microsoft.onmicrosoft.com%2fs%2fbLeGBltMW0qI0WRLsTqx2A2?ismsaljsauthenabled&data=05%7c02%7cv-mchougule%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838215467554%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=i4Fn6qPJw4eg3Ny6En0%2fPnIqw%2BpJpHuT%2f%2BZH2hE%2fRTA%3D&reserved=0)."

"[Check out Accessibility Insights](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0)! - Identify accessibility bugs before check-in and make bug fixing faster and easier."

"Contact [CAI Accessibility Support Channel](https://teams.microsoft.com/l/channel/19:DeCtYeREyQHw5jYsz9PPc3hpVw48m-tOlTkQ1jEn0eM1%40thread.tacv2/CAI%20Accessibility%20Support%20Channel?groupId=f5eb428c-9e6e-4c16-8143-33fe3961a9ae&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) for Accessibility queries on design, development, test, usability or any processes"

Please do not close this bug. This bug should only be closed by Trusted Tester after verification.

### GitHub Tags:
#A11yTCS; #A11yMAS; #A11yAuto; #Win11; #DesktopWeb; #Microsoft MakeCode; #E2E_MicrosoftMakeCode_Web_May2026; #FTP; #ChromiumEdge; #Contrast (minimum); #A11ySev2; #WCAG1.4.3; #AI4W; #ES Chat; #copilot:agent=a11yagent; #color-contrast;

### Environment Details:
Application Name: Microsoft MakeCode
URL: https://makecode.microbit.org/beta
windows Edition: Windows 11 Enterprise Insider Preview(24H2)
OS Build: 26544.1000

### Repro Steps:
1. Launch application URL : https://makecode.microbit.org
2. Home page will open.
3. Tab to "New Project" option and select it. Editor screen will open.
5. Tab to "Basic, Input, Music, Led, more, Radio" controls.
6. Run accessibility insights for web.
7. Observe the issue.

### Issue: 
Ensure the contrast between foreground and background colors meets WCAG 2 AA minimum contrast ratio thresholds (color-contrast - https://accessibilityinsights.io/info-examples/web/color-contrast)

### Target application:
 Microsoft MakeCode for micro:bit - https://makecode.microbit.org/beta#editor

### Element path:
 #basic\.label

### Snippet:
 <span id="basic.label" class="blocklyTreeLabel">Basic</span>

Related paths: 
  #blockly-0

### How to fix: 
Fix any of the following:
  Element has insufficient color contrast of 3.23 (foreground color: #1e90ff, background color: #ffffff, font size: 13.8pt (18.4px), font weight: normal). Expected contrast ratio of 4.5:1

**Environment:** Microsoft Edge version 148.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.46.0 (axe-core 4.10.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### Attachment:

<img width="959" height="481" alt="Image" src="https://github.com/user-attachments/assets/0ca6ae5c-1400-4c4e-a956-45845b4012e3" />

**Comment by riknoll** [comment]
unfortunately this cannot be changed because it has been shipped for a long time and changing these block colors would significantly alter the appearance of the editor. makecode is primarily used in education and many teachers take screenshots of our editor for their course materials that they distribute to students. we have received consistent feedback that making visual changes causes a great deal of hardship for these teachers because students get confused while following screenshots that do not exactly match what they see on the screen.

we offer a high contrast theme that can be accessed in the settings menu that works around this issue. in high contrast mode, the text is rendered as white on black:

<img width="220" height="413" alt="Image" src="https://github.com/user-attachments/assets/48dd6b60-c888-441a-bb74-0db84ebed1b6" />

the site also detects the OS high contrast settings and automatically uses this theme when the user has high contrast enabled on their computer
